src/backends/torch/torchlib.h

```diff
@@ -90,13 +94,24 @@ namespace dd
                  TorchDataset &dataset,
                  int batch_size, APIData &out);
 
+	  /**
+	   * \brief	create recurrent.prototxt using api data
+	   * @param ad input apidata
+	   * @param inputc input connector
+	   * @param net_param prototxt infos
+	   */
+	void configure_recurrent_template(const APIData &ad, TInputConnectorStrategy &inputc, caffe::NetParameter &net_param);
```

Comment:
Do we want Caffe stuff to leak into `torchlib` ? Looking at the function internals, it seems it does stuff that could be moved outside `torchlib`, I let you confirm this. One way to have it outside could be to move it to `net_generator`. Another way, possibly a bit more complicated, could be to use our own copy of `NetParameter` from outside Caffe since it's just protobuf, as far as I understand. What are your thoughts ?

Reply:
I am not sure to understand your point.
Indeed it is already a copy of NetParameter as we include only caffe.pb.h and nothing else from caffe. (I generally build this backend w/o caffe) Using orignal caffe.proto makes NetParameter be caffe::NetParameter, and I think it is better to use only this one copy of caffe.proto (the same is used for every backend), protobuf stuff is already too complex.

Reply:
Having caffe stuff here is confusing, I get that the `caffe::NetParameter` is good enough to not reinvent stuff. 

So I'm more in favor in moving this to net_generator to hide the implementation detail that currently uses some `net_caffe` internals.

Reply:
+1 for moving it to another file

Reply:
moved to net_generator/net_caffe_recurrent.[h|cc]